libp2p/src/builder.rs

```diff
@@ -0,0 +1,1795 @@
+    not(target_arch = "wasm32"),
+    feature = "tcp",
+    any(feature = "tls", feature = "noise")
+))]
+impl<Provider> SwarmBuilder<Provider, TcpTlsPhase> {
+    #[cfg(feature = "tls")]
+    pub fn with_tls(
+        self,
+    ) -> Result<SwarmBuilder<Provider, TcpNoisePhase<libp2p_tls::Config>>, AuthenticationError>
+    {
+        Ok(SwarmBuilder {
+            phase: TcpNoisePhase {
+                tcp_config: self.phase.tcp_config,
+                tls_config: libp2p_tls::Config::new(&self.keypair)
+                    .map_err(|e| AuthenticationError(e.into()))?,
+                phantom: PhantomData,
+            },
+            keypair: self.keypair,
+            phantom: PhantomData,
+        })
+    }
+
+    fn without_tls(self) -> SwarmBuilder<Provider, TcpNoisePhase<WithoutTls>> {
+        SwarmBuilder {
+            keypair: self.keypair,
+            phantom: PhantomData,
+            phase: TcpNoisePhase {
+                tcp_config: self.phase.tcp_config,
+                tls_config: WithoutTls {},
+                phantom: PhantomData,
+            },
+        }
+    }
+}
+
+// Shortcuts
+#[cfg(all(
+    not(target_arch = "wasm32"),
+    feature = "tcp",
+    feature = "noise",
+    feature = "async-std"
+))]
+impl SwarmBuilder<AsyncStd, TcpTlsPhase> {
+    pub fn with_noise(
+        self,
+    ) -> Result<
+        SwarmBuilder<AsyncStd, QuicPhase<impl AuthenticatedMultiplexedTransport>>,
+        AuthenticationError,
+    > {
+        self.without_tls().with_noise()
+    }
+}
+#[cfg(all(
+    not(target_arch = "wasm32"),
+    feature = "tcp",
+    feature = "noise",
+    feature = "tokio"
+))]
+impl SwarmBuilder<Tokio, TcpTlsPhase> {
+    pub fn with_noise(
+        self,
+    ) -> Result<
+        SwarmBuilder<Tokio, QuicPhase<impl AuthenticatedMultiplexedTransport>>,
+        AuthenticationError,
+    > {
+        self.without_tls().with_noise()
+    }
+}
+
+#[cfg(all(
+    not(target_arch = "wasm32"),
+    feature = "tcp",
+    any(feature = "tls", feature = "noise")
+))]
+pub struct TcpNoisePhase<A> {
+    tcp_config: libp2p_tcp::Config,
+    tls_config: A,
+    phantom: PhantomData<A>,
+}
+
+#[cfg(all(
+    not(target_arch = "wasm32"),
+    feature = "tcp",
+    feature = "yamux",
+    any(feature = "tls", feature = "noise")
+))]
+macro_rules! construct_quic_builder {
+    ($self:ident, $tcp:ident, $auth:expr) => {
+        SwarmBuilder {
+            phase: QuicPhase {
+                transport: libp2p_tcp::$tcp::Transport::new($self.phase.tcp_config)
+                    .upgrade(libp2p_core::upgrade::Version::V1Lazy)
+                    .authenticate($auth)
+                    .multiplex(libp2p_yamux::Config::default())
+                    .map(|(p, c), _| (p, StreamMuxerBox::new(c))),
+            },
+            keypair: $self.keypair,
+            phantom: PhantomData,
+        }
+    };
+}
+
+macro_rules! impl_tcp_noise_builder {
+    ($providerKebabCase:literal, $providerCamelCase:ident, $tcp:ident) => {
```

Comment:
```suggestion
    ($providerKebabCase:literal, $providerPascalCase:ident, $tcp:ident) => {
```

`camelCase` != `PascalCase`.

Reply:
83e3674dc